frameworks/non-keyed/kobold/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+    let is_in_danger = class!("danger" if state.selected == Some(row.id));
+
+    bind! { state:
+        let remove = move |_| state.remove(num);
+        let select = move |_| state.select(row.id);
+    }
+
+    view! {
+        <tr.{is_in_danger}>
+            <td class="col-md-1">{row.id.to_string()}</td>
+            <td class="col-md-4"><a onclick={select}>{row.label}</a></td>
+            <td class="col-md-1"><a onclick={remove}><span class="glyphicon glyphicon-remove" aria-hidden="true"></span></a></td>
+            <td class="col-md-6"/>
+        </tr>
+    }
+}
+
+enum ButtonAction {
+    Run,
+    RunLots,
+    Add,
+    Update,
+    Clear,
+    Swap,
+}
+
+#[component]
+fn Button(action: ButtonAction, state: &Hook<State>) -> impl View {
+    let (id, text) = match action {
+        ButtonAction::Run => ("run", "Create 1,000 rows"),
+        ButtonAction::RunLots => ("runlots", "Create 10,000 rows"),
+        ButtonAction::Add => ("add", "Append 1,000 rows"),
+        ButtonAction::Update => ("update", "Update every 10th row"),
+        ButtonAction::Clear => ("clear", "Clear"),
+        ButtonAction::Swap => ("swaprows", "Swap Rows"),
+    };
+
+    bind! { state:
+        let onclick = move |_| match action {
+            ButtonAction::Run => state.run(),
+            ButtonAction::RunLots => state.run_lots(),
+            ButtonAction::Add => state.add(),
+            ButtonAction::Update => state.update(),
+            ButtonAction::Clear => state.clear(),
+            ButtonAction::Swap => state.swap(),
+        };
+    }
+
+    view! {
+        <div class="col-sm-6 smallpad">
+        <button
+            id={id}
+            class="btn btn-primary btn-block"
+            type="button"
+            {onclick}
+        >
+            { text }
+        </button>
+    </div>
+    }
+}
+
+#[component]
+fn App() -> impl View {
+    stateful(State::new, |state| {
+        view! {
+            <div class="container">
+            <div class="jumbotron">
+                <div class="row">
+                    <div class="col-md-6"><h1>"Kobold"</h1></div>
+                    <div class="col-md-6">
+                        <div class="row">
+                            <Button {state} action={ButtonAction::Run} />
+                            <Button {state} action={ButtonAction::RunLots} />
+                            <Button {state} action={ButtonAction::Add} />
+                            <Button {state} action={ButtonAction::Update} />
+                            <Button {state} action={ButtonAction::Clear} />
+                            <Button {state} action={ButtonAction::Swap} />
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <table class="table table-hover table-striped test-data">
+                <tbody>
+                { for state.rows.iter().enumerate().map(|(i,l)| view! { <Row {state}  num={i} row={l.to_owned()} />}) }
```

Comment:
```suggestion
                { for state.rows.iter().enumerate().map(|(num,row)| view! { <Row {state}  {num} {row} /> }) }
```
No need to clone `RowData`

Reply:
[Committed](https://github.com/krausest/js-framework-benchmark/pull/1239/commits/7bc250c09eb3fe71ca83175c3749975ddabe606d)